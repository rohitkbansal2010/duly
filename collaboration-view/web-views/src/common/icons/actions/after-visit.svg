<svg width="25" height="27" viewBox="0 0 25 27" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M13.0527 2H5C3.34315 2 2 3.34315 2 5V22C2 23.6569 3.34314 25 5 25H13.0527" stroke="white" stroke-width="2.5" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M10.1055 13.871H23.0002M23.0002 13.871L17.1055 7.93555M23.0002 13.871L17.4739 19.8065" stroke="white" stroke-width="2.5" stroke-linecap="round"/>
</svg>
