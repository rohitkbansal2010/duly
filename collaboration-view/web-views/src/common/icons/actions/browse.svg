<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M20.4678 8.96715C20.4678 8.96715 20.4678 8.96715 20.4678 8.92202C19.8324 7.18556 18.6795 5.68634 17.165 4.62717C15.6505 3.568 13.8475 3 12 3C10.1525 3 8.3495 3.568 6.83501 4.62717C5.32052 5.68634 4.1676 7.18556 3.5322 8.92202C3.5322 8.92202 3.5322 8.92202 3.5322 8.96715C2.8226 10.9267 2.8226 13.0733 3.5322 15.0328C3.5322 15.0328 3.5322 15.0328 3.5322 15.078C4.1676 16.8144 5.32052 18.3137 6.83501 19.3728C8.3495 20.432 10.1525 21 12 21C13.8475 21 15.6505 20.432 17.165 19.3728C18.6795 18.3137 19.8324 16.8144 20.4678 15.078C20.4678 15.078 20.4678 15.078 20.4678 15.0328C21.1774 13.0733 21.1774 10.9267 20.4678 8.96715ZM5.00211 13.8053C4.6894 12.6221 4.6894 11.3779 5.00211 10.1947H6.67944C6.53518 11.3939 6.53518 12.6061 6.67944 13.8053H5.00211ZM5.74158 15.6105H7.00409C7.21575 16.4155 7.5183 17.1938 7.90588 17.9303C7.02149 17.3273 6.2822 16.5349 5.74158 15.6105ZM7.00409 8.38947H5.74158C6.27442 7.46796 7.00426 6.67579 7.87882 6.0697C7.50057 6.80737 7.20712 7.5856 7.00409 8.38947ZM11.0802 18.9503C9.97257 18.1367 9.19467 16.9512 8.88883 15.6105H11.0802V18.9503ZM11.0802 13.8053H8.50106C8.33277 12.6076 8.33277 11.3924 8.50106 10.1947H11.0802V13.8053ZM11.0802 8.38947H8.88883C9.19467 7.04885 9.97257 5.86329 11.0802 5.04973V8.38947ZM18.2223 8.38947H16.9598C16.7482 7.58446 16.4456 6.80619 16.0581 6.0697C16.9424 6.67272 17.6817 7.46512 18.2223 8.38947ZM12.8838 5.04973C13.9914 5.86329 14.7693 7.04885 15.0751 8.38947H12.8838V5.04973ZM12.8838 18.9503V15.6105H15.0751C14.7693 16.9512 13.9914 18.1367 12.8838 18.9503ZM15.4629 13.8053H12.8838V10.1947H15.4629C15.6312 11.3924 15.6312 12.6076 15.4629 13.8053ZM16.0851 17.9303C16.4727 17.1938 16.7752 16.4155 16.9869 15.6105H18.2494C17.7088 16.5349 16.9695 17.3273 16.0851 17.9303ZM18.9618 13.8053H17.2845C17.3578 13.2063 17.394 12.6034 17.3927 12C17.3937 11.3966 17.3576 10.7937 17.2845 10.1947H18.9618C19.2745 11.3779 19.2745 12.6221 18.9618 13.8053Z" fill="#002855"/>
</svg>
