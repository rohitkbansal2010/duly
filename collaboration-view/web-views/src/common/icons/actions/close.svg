<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M13.409 11.9991L19.7044 5.71303C19.8926 5.52485 19.9983 5.26961 19.9983 5.00347C19.9983 4.73733 19.8926 4.4821 19.7044 4.29391C19.5162 4.10572 19.261 4 18.9949 4C18.7288 4 18.4736 4.10572 18.2854 4.29391L12 10.59L5.71456 4.29391C5.52639 4.10572 5.27118 4 5.00507 4C4.73896 4 4.48375 4.10572 4.29558 4.29391C4.10742 4.4821 4.00171 4.73733 4.00171 5.00347C4.00171 5.26961 4.10742 5.52485 4.29558 5.71303L10.591 11.9991L4.29558 18.2853C4.20192 18.3782 4.12758 18.4887 4.07685 18.6105C4.02612 18.7323 4 18.8629 4 18.9948C4 19.1268 4.02612 19.2574 4.07685 19.3792C4.12758 19.5009 4.20192 19.6115 4.29558 19.7044C4.38848 19.7981 4.499 19.8724 4.62077 19.9231C4.74254 19.9739 4.87315 20 5.00507 20C5.13699 20 5.2676 19.9739 5.38937 19.9231C5.51114 19.8724 5.62166 19.7981 5.71456 19.7044L12 13.4083L18.2854 19.7044C18.3783 19.7981 18.4889 19.8724 18.6106 19.9231C18.7324 19.9739 18.863 20 18.9949 20C19.1268 20 19.2575 19.9739 19.3792 19.9231C19.501 19.8724 19.6115 19.7981 19.7044 19.7044C19.7981 19.6115 19.8724 19.5009 19.9232 19.3792C19.9739 19.2574 20 19.1268 20 18.9948C20 18.8629 19.9739 18.7323 19.9232 18.6105C19.8724 18.4887 19.7981 18.3782 19.7044 18.2853L13.409 11.9991Z" fill="#002855"/>
</svg>
