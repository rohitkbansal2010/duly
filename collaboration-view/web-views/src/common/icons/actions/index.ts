import addIcon from './add.svg';
import AfterVisitLinkIcon from './after-visit.svg';
import attachIcon from './attach.svg';
import browseIcon from './browse.svg';
import closeIcon from './close.svg';
import cropIcon from './crop.svg';
import downloadIcon from './download.svg';
import editIcon from './edit.svg';
import expandSolidIcon from './expand-solid.svg';
import expandIcon from './expand.svg';
import ExportPdf from './export-pdf.svg';
import favoriteIcon from './favorite.svg';
import filterOutlinedIcon from './filter-outlined.svg';
import filterSolidIcon from './filter-solid.svg';
import FontIconBlue from './font-icon-blue.svg';
import FontIconSkyblue from './font-icon-skyblue.svg';
import lineToolIcon from './line-tool.svg';
import Line from './Line.svg';
import messageIcon from './message.svg';
import moreIcon from './more.svg';
import noteIcon from './note.svg';
import printIcon from './print.svg';
import reflectIcon from './reflect.svg';
import reviewVitalsIcon from './review-vitals.svg';
import searchIcon from './search.svg';
import sendIcon from './send.svg';
import shapeToolIcon from './shape-tool.svg';
import textToolIcon from './text-tool.svg';
import updateIcon from './update.svg';
import uploadIcon from './upload.svg';
import viewPdfIcon from './view-pdf.svg';
import voiceRecorderIcon from './voice-recorder.svg';

export {
  reviewVitalsIcon,
  viewPdfIcon,
  voiceRecorderIcon,
  textToolIcon,
  updateIcon,
  uploadIcon,
  shapeToolIcon,
  searchIcon,
  sendIcon,
  printIcon,
  reflectIcon,
  addIcon,
  noteIcon,
  lineToolIcon,
  messageIcon,
  moreIcon,
  filterSolidIcon,
  attachIcon,
  filterOutlinedIcon,
  browseIcon,
  closeIcon,
  cropIcon,
  downloadIcon,
  favoriteIcon,
  editIcon,
  expandIcon,
  expandSolidIcon,
  FontIconSkyblue,
  FontIconBlue,
  AfterVisitLinkIcon,
  ExportPdf,
  Line
};
