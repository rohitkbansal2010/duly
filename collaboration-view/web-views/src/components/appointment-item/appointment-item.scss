@import "~@styles/colors.scss";
@import "~@styles/main.scss";
@import "~@styles/typography.scss";

.appointmentContainer {
  z-index: 20;
  width: 0 !important;
}

.ellipsis {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.dots {
  overflow: hidden;
  color: $default-text;
  font-size: 1rem !important;
  line-height: 1.5rem !important;
  white-space: nowrap;
  text-align: center;
  @include nexa-extra-bold-highlight;
}

.btgRect {
  display: inline-block;
  width: 5.75rem;
  height: 1.3125rem;
  margin-right: 0.25rem;
  color: $black;
  vertical-align: sub;
  background-color: $black;
  border-radius: 0.5rem;
}

.appointmentCard {
  position: relative;
  display: flex !important;
  flex-direction: row !important;
  border: 0 !important;
  border-radius: 0 !important;
  cursor: pointer;

  &NewPatient {
    position: absolute;
    top: 0.25rem;
    right: 0.7rem;
    display: flex;
    align-items: center !important;
    justify-content: center !important;
    width: 1.5rem;
    height: 1.5rem;
    object-fit: contain;
    background: linear-gradient(
      90deg,
      convertToRGBA($dupage-light-blue, 0.1) 0.7%,
      convertToRGBA($dupage-light-blue, 0.8) 99.11%
    );
    border-radius: 50%;

    & img {
      width: 1rem;
      height: 0.75rem;
    }
  }

  &BarLeft {
    width: 0.63rem;
    height: auto;
    background-color: $dupage-light-blue;
    border-radius: 0.5rem 0 0 0.5rem;
  }

  &Body {
    flex: 1 !important;
    margin: 0 !important;
    padding: 0.25rem 0.6rem 0.56rem 0.75rem !important;
    overflow: hidden;
    background: linear-gradient(
      90deg,
      convertToRGBA($dupage-light-blue, 0.1) 0.7%,
      convertToRGBA($dupage-light-blue, 0.8) 99.11%
    );

    &Content {
      &Title {
        display: flex !important;
        align-items: center !important;
        margin: 0 !important;

        & img {
          width: 0.81rem;
          height: 0.56rem;
          margin-left: 0.31rem;
        }
      }

      &HumanName,
      &ServiceType {
        color: $default-text;
        font-size: 1rem !important;
        line-height: 1.5rem !important;
      }

      &HumanName {
        @include nexa-extra-bold-highlight;
      }

      &ServiceType {
        @include nexa-regular-highlight;
      }

      &Practitioner {
        margin: 0 !important;
        color: $default-text;
        font-size: 1rem !important;
        line-height: 1.5rem !important;
        @include nexa-regular-highlight;
      }

      &Time {
        color: $dark-gray;
        font-size: 1rem !important;
        line-height: 1.5rem !important;
        @include nexa-regular-highlight;
      }
    }
  }
}
