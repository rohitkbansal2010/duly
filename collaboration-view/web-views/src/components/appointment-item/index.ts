export { AppointmentItem } from './appointment-item';
