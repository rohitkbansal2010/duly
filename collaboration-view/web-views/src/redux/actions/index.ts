export * from './ui';
export * from './locale';
export * from './types';
export * from './user';
export * from './app-state';
export * from './todays-appointments';
export * from './current-appointment';
export * from './overview-widgets';
export * from './site';
