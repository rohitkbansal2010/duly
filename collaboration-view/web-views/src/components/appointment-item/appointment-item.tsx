import { useEffect, useRef, useState } from 'react';
import { Card, Col } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';

import {
  appointmentItemUI,
  HEIGHT_HOUR_SEGMENT,
  MINUTES_PER_HOUR,
  PADDING_BOTTOM_SCHEDULER_SCALE,
  SEGMENT_IN_MINUTES
} from '@constants';
import { AppointmentType } from '@enums';
import { addUserWhiteIcon, videoCameraFillDarkBlueIcon } from '@icons';
import { setPickedAppointmentId, showBTGModal } from '@redux/actions';
import { RootState } from '@redux/reducers';
import { AppDispatch } from '@redux/store';
import { AppointmentData } from '@types';
import {
  calcDurationInMinutes,
  calcTopOfAppointment,
  isTitleNotEmpty,
  writeHumanName,
  writeTimeInterval
} from '@utils';

import styles from './appointment-item.scss';

export type AppointmentItemProps = AppointmentData & {
  startTimeGroups: number;
};

export const AppointmentItem = ({
  id,
  title,
  type,
  timeSlot: {
    startTime,
    endTime,
  },
  patient: {
    patientGeneralInfo: {
      humanName: {
        familyName: patientFamilyName,
        givenNames,
      },
    },
    isNewPatient,
  },
  practitioner: {
    humanName: {
      familyName: practitionerFamilyName,
      prefixes = [],
    },
  },
  isProtectedByBtg,
  startTimeGroups,
}: AppointmentItemProps) => {
  const [ height, setHeight ] = useState<number>(0);

  const dispatch: AppDispatch = useDispatch();

  const duration = calcDurationInMinutes(new Date(startTime), new Date(endTime));
  const isDurationQuarter = duration === SEGMENT_IN_MINUTES;

  useEffect(() => {
    setHeight((duration / MINUTES_PER_HOUR) * HEIGHT_HOUR_SEGMENT - PADDING_BOTTOM_SCHEDULER_SCALE);
  }, [ startTime, endTime, duration ]);

  const ellipsisParentRef = useRef<HTMLDivElement>(null);
  const ellipsisChildRef = useRef<HTMLSpanElement>(null);
  const newPatientIconRef = useRef<HTMLSpanElement>(null);

  const [ isSpaceForNewPatientIcon, setIsSpaceForNewPatientIcon ] = useState(true);
  const [ isSpaceForText, setIsSpaceForText ] = useState(true);

  useEffect(() => {
    if (ellipsisParentRef.current && ellipsisChildRef.current && newPatientIconRef.current) {
      const diff = ellipsisParentRef.current.offsetWidth - ellipsisChildRef.current.offsetWidth;
      const iconWidth = newPatientIconRef.current.offsetWidth;
      setIsSpaceForNewPatientIcon(diff > iconWidth);
    } else {
      setIsSpaceForNewPatientIcon(true);
    }
  }, [ ellipsisParentRef, ellipsisChildRef, newPatientIconRef ]);

  const currentScreenWidth = useSelector(({ UI }: RootState) =>
    UI.width);

  useEffect(() => {
    if (ellipsisParentRef.current && currentScreenWidth) {
      const minWidth =
        currentScreenWidth / appointmentItemUI.BASIC_SCREEN_WIDTH * appointmentItemUI.MIN_WIDTH;
      setIsSpaceForText(ellipsisParentRef.current.offsetWidth > minWidth);
    } else {
      setIsSpaceForText(true);
    }
  }, [ ellipsisParentRef, currentScreenWidth ]);

  const handleCardClick = () => {
    isProtectedByBtg ? dispatch(showBTGModal()) : dispatch(setPickedAppointmentId(id));
  };

  return (
    <Col className={styles.appointmentContainer}>
      <Card
        className={styles.appointmentCard}
        style={{
          top: `${calcTopOfAppointment(new Date(startTime), new Date(startTimeGroups))}rem`,
          height: `${height}rem`,
        }}
      >
        {isNewPatient && isSpaceForNewPatientIcon && (
        <span ref={newPatientIconRef} className={styles.appointmentCardNewPatient}>
          <img src={addUserWhiteIcon} data-testid="appointment-item__new-patient-icon" alt="White add user" />
        </span>
				)}
        <div
          className={styles.appointmentCardBarLeft}
        />
        <Card.Body
          className={styles.appointmentCardBody}
          onClick={handleCardClick}
        >
          {isSpaceForText ? (
            <div className={styles.appointmentCardBodyContent}>
              <Card.Title
                ref={ellipsisParentRef}
                className={styles.appointmentCardBodyContentTitle}
              >
                <span ref={ellipsisChildRef} className={styles.ellipsis}>
                  <span
                    data-testid="appointment-item__patient-name"
                    className={styles.appointmentCardBodyContentHumanName}
                  >
                    {isProtectedByBtg ?
                      <span className={styles.btgRect} /> :
											writeHumanName(givenNames, patientFamilyName)
										}
                  </span>
                  {isTitleNotEmpty(title) && (
                  <>
                    <span className={styles.appointmentCardBodyContentHumanName}>
                      {isProtectedByBtg || ', '}
                    </span>
                    <span className={styles.appointmentCardBodyContentServiceType}>
                      {title}
                    </span>
                  </>
									)}
                  {type === AppointmentType.TELEHEALTH && (
                  <img
                    src={videoCameraFillDarkBlueIcon}
                    data-testid="appointment-item__telehealth-icon"
                    alt="Filled dark blue videocam"
                  />
									)}
                </span>
              </Card.Title>
              <div className={!isDurationQuarter ? 'd-flex flex-column' : styles.ellipsis}>
                <span
                  data-testid="appointment-item__practitioner-card"
                  className={`${styles.appointmentCardBodyContentPractitioner} ${styles.ellipsis}`}
                >
                  {`${writeHumanName(prefixes, practitionerFamilyName)}`.trim()}
                  {isDurationQuarter ? ', ' : ''}
                </span>
                <span className={`${styles.appointmentCardBodyContentTime} ${styles.ellipsis}`}>
                  {writeTimeInterval(new Date(startTime), new Date(endTime))}
                </span>
              </div>
            </div>
				) : (<div className={styles.dots}>...</div>)}
        </Card.Body>
      </Card>
    </Col>
  );
};
