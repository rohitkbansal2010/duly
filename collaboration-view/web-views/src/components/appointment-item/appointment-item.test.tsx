import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
// eslint-disable-next-line no-use-before-define
import React from 'react';
import { Card } from 'react-bootstrap';

import { AppointmentItem, AppointmentItemProps } from '@components/appointment-item/appointment-item';
import {
  AppointmentStatus, AppointmentType, CareTeamMemberRole, PractitionerRole 
} from '@enums';
import { setPickedAppointmentId } from '@redux/actions';

const mockDispatch = jest.fn();
jest.mock('react-redux', () =>
  ({
    ...jest.requireActual('react-redux'),
    useDispatch: () =>
      mockDispatch,
    useSelector: jest.fn().mockImplementation(() => 
      ({ width: 1920 })),
  }));
	

jest.spyOn(Date.prototype, 'getTimezoneOffset')
  .mockImplementation(() =>
    0);

jest.mock('../../common/utils/appointment', () =>
  ({
    ...jest.requireActual('../../common/utils/appointment'),
    calcTopOfAppointment: jest.fn().mockReturnValue(0),
  }));

jest.mock('../../common/utils/date', () =>
  ({
    ...jest.requireActual('../../common/utils/date'),
    writeTimeInterval: jest.fn().mockReturnValue('0:00 - 0:00 pm'),
  }));

const emptyAppointment = {
  id: '1',
  title: '',
  type: AppointmentType.TELEHEALTH,
  status: AppointmentStatus.SCHEDULED,
  timeSlot: {
    startTime: '',
    endTime: '',
  },
  location: { title: '' },
  patient: {
    patientGeneralInfo: {
      humanName: {
        familyName: '',
        givenNames: [],
      },
      id: '1',
    },
    isNewPatient: true,
  },
  practitioner: {
    humanName: {
      familyName: '',
      givenNames: [],
    },
    id: '1',
    role: CareTeamMemberRole.DOCTOR,
  },
  isProtectedByBtg: false,
  startTimeGroups: 1,
};

describe('AppointmentItem', () => {
  let Component: ReturnType<typeof mount>;
  const props: AppointmentItemProps = {
    id: '1',
    title: 'test appointment',
    type: AppointmentType.TELEHEALTH,
    status: AppointmentStatus.SCHEDULED,
    timeSlot: {
      startTime: '2022-01-13T00:00:00',
      endTime: '2022-01-13T01:00:00',
    },
    patient: {
      patientGeneralInfo: {
        humanName: {
          familyName: 'second name',
          givenNames: [],
        },
        id: '1',
      },
      isNewPatient: true,
    },
    practitioner: {
      humanName: {
        familyName: 'practioner name',
        givenNames: [],
      },
      id: '1',
      role: CareTeamMemberRole.DOCTOR,
    },
    isProtectedByBtg: false,
    startTimeGroups: 1,
  };

  beforeEach(() => {
    jest.spyOn(React, 'useEffect').mockImplementation(jest.fn());
    Component = mount(
      <AppointmentItem {...props} />
    );
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('should render correctly', () => {
    expect(toJson(Component)).toMatchSnapshot();
  });

  it('should render telehealth image if appointment is telehealth', () => {
    expect(Component.find({ 'data-testid': 'appointment-item__telehealth-icon' })
      .exists()).toBe(true);
  });

  it('should render new patient image if patient is new', () => {
    expect(Component.find({ 'data-testid': 'appointment-item__new-patient-icon' })
      .exists()).toBe(true);
  });

  it('should dispatch correct action by clicking appointment', () => {
    const wrapper = Component.find(Card.Body);
    wrapper.simulate('click');
    expect(mockDispatch).toHaveBeenCalledWith(setPickedAppointmentId(props.id));
  });

  it('should not render image if appointment type is not telehealth', () => {
    const wrapper = mount(
      <AppointmentItem {...props} type={AppointmentType.ON_SITE} />
    );

    expect(wrapper.find({ 'data-testid': 'appointment-item__telehealth-icon' })
      .exists()).toBe(false);
  });

  it('should not apply column align class if duration is 15 minutes', () => {
    const timeSlot = {
      startTime: '2022-01-13T00:00:00Z',
      endTime: '2022-01-13T00:15:00Z',
    };

    const wrapper = mount(
      <AppointmentItem {...props} timeSlot={timeSlot} />
    );

    expect(wrapper.find('.d-flex flex-column').exists()).toBe(false);
  });

  it('should render correctly appointment with empty data', () => {
    const wrapper = mount(
      <AppointmentItem {...emptyAppointment} />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly given names of the patient', () => {
    const patient = {
      patientGeneralInfo: {
        humanName: {
          familyName: 'patient name',
          givenNames: [ 'patientName1', 'patientName2' ],
        },
        id: '1',
      },
      isNewPatient: true,
    };

    const wrapper = mount(<AppointmentItem {...props} patient={patient}/>);

    const patientName = wrapper.find({ 'data-testid': 'appointment-item__patient-name' }).text();
    expect(patientName.includes(patient.patientGeneralInfo.humanName.familyName)).toBe(true);

    patient.patientGeneralInfo.humanName.givenNames.forEach((name) => {
      expect(patientName.includes(name));
    });
  });

  it('should render correctly given names of the practitioner', () => {
    const practitioner = {
      humanName: {
        familyName: 'practioner name',
        givenNames: [ 'givenname1', 'givenname2' ],
      },
      id: '1',
      role: PractitionerRole.PCP,
    };

    const wrapper = mount(<AppointmentItem {...props} practitioner={practitioner} />);

    const practitionerInfo = wrapper.
      find({ 'data-testid': 'appointment-item__practitioner-card' }).text();
    expect(practitionerInfo.includes(practitioner.humanName.familyName)).toBe(true);

    practitioner.humanName.givenNames.forEach((name) => {
      expect(practitionerInfo.includes(name));
    });
  });
});
