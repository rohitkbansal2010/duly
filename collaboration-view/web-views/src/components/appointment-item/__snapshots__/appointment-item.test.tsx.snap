// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentItem should render correctly 1`] = `
<AppointmentItem
  id="1"
  isProtectedByBtg={false}
  patient={
    Object {
      "isNewPatient": true,
      "patientGeneralInfo": Object {
        "humanName": Object {
          "familyName": "second name",
          "givenNames": Array [],
        },
        "id": "1",
      },
    }
  }
  practitioner={
    Object {
      "humanName": Object {
        "familyName": "practioner name",
        "givenNames": Array [],
      },
      "id": "1",
      "role": "Doctor",
    }
  }
  startTimeGroups={1}
  status="Scheduled"
  timeSlot={
    Object {
      "endTime": "2022-01-13T01:00:00",
      "startTime": "2022-01-13T00:00:00",
    }
  }
  title="test appointment"
  type="Telehealth"
>
  <Col
    className="appointmentContainer"
  >
    <div
      className="appointmentContainer col"
    >
      <Card
        body={false}
        className="appointmentCard"
        style={
          Object {
            "height": "0rem",
            "top": "0rem",
          }
        }
      >
        <div
          className="appointmentCard card"
          style={
            Object {
              "height": "0rem",
              "top": "0rem",
            }
          }
        >
          <span
            className="appointmentCardNewPatient"
          >
            <img
              alt="White add user"
              data-testid="appointment-item__new-patient-icon"
              src="add-user-white.svg"
            />
          </span>
          <div
            className="appointmentCardBarLeft"
          />
          <CardBody
            className="appointmentCardBody"
            onClick={[Function]}
          >
            <div
              className="appointmentCardBody card-body"
              onClick={[Function]}
            >
              <div
                className="appointmentCardBodyContent"
              >
                <CardTitle
                  className="appointmentCardBodyContentTitle"
                >
                  <ForwardRef
                    className="appointmentCardBodyContentTitle card-title"
                  >
                    <div
                      className="appointmentCardBodyContentTitle card-title h5"
                    >
                      <span
                        className="ellipsis"
                      >
                        <span
                          className="appointmentCardBodyContentHumanName"
                          data-testid="appointment-item__patient-name"
                        >
                           second name
                        </span>
                        <span
                          className="appointmentCardBodyContentHumanName"
                        >
                          , 
                        </span>
                        <span
                          className="appointmentCardBodyContentServiceType"
                        >
                          test appointment
                        </span>
                        <img
                          alt="Filled dark blue videocam"
                          data-testid="appointment-item__telehealth-icon"
                          src="video-camera-fill-dark-blue.svg"
                        />
                      </span>
                    </div>
                  </ForwardRef>
                </CardTitle>
                <div
                  className="d-flex flex-column"
                >
                  <span
                    className="appointmentCardBodyContentPractitioner ellipsis"
                    data-testid="appointment-item__practitioner-card"
                  >
                    practioner name
                  </span>
                  <span
                    className="appointmentCardBodyContentTime ellipsis"
                  >
                    0:00 - 0:00 pm
                  </span>
                </div>
              </div>
            </div>
          </CardBody>
        </div>
      </Card>
    </div>
  </Col>
</AppointmentItem>
`;

exports[`AppointmentItem should render correctly appointment with empty data 1`] = `
<AppointmentItem
  id="1"
  isProtectedByBtg={false}
  location={
    Object {
      "title": "",
    }
  }
  patient={
    Object {
      "isNewPatient": true,
      "patientGeneralInfo": Object {
        "humanName": Object {
          "familyName": "",
          "givenNames": Array [],
        },
        "id": "1",
      },
    }
  }
  practitioner={
    Object {
      "humanName": Object {
        "familyName": "",
        "givenNames": Array [],
      },
      "id": "1",
      "role": "Doctor",
    }
  }
  startTimeGroups={1}
  status="Scheduled"
  timeSlot={
    Object {
      "endTime": "",
      "startTime": "",
    }
  }
  title=""
  type="Telehealth"
>
  <Col
    className="appointmentContainer"
  >
    <div
      className="appointmentContainer col"
    >
      <Card
        body={false}
        className="appointmentCard"
        style={
          Object {
            "height": "0rem",
            "top": "0rem",
          }
        }
      >
        <div
          className="appointmentCard card"
          style={
            Object {
              "height": "0rem",
              "top": "0rem",
            }
          }
        >
          <span
            className="appointmentCardNewPatient"
          >
            <img
              alt="White add user"
              data-testid="appointment-item__new-patient-icon"
              src="add-user-white.svg"
            />
          </span>
          <div
            className="appointmentCardBarLeft"
          />
          <CardBody
            className="appointmentCardBody"
            onClick={[Function]}
          >
            <div
              className="appointmentCardBody card-body"
              onClick={[Function]}
            >
              <div
                className="appointmentCardBodyContent"
              >
                <CardTitle
                  className="appointmentCardBodyContentTitle"
                >
                  <ForwardRef
                    className="appointmentCardBodyContentTitle card-title"
                  >
                    <div
                      className="appointmentCardBodyContentTitle card-title h5"
                    >
                      <span
                        className="ellipsis"
                      >
                        <span
                          className="appointmentCardBodyContentHumanName"
                          data-testid="appointment-item__patient-name"
                        >
                           
                        </span>
                        <img
                          alt="Filled dark blue videocam"
                          data-testid="appointment-item__telehealth-icon"
                          src="video-camera-fill-dark-blue.svg"
                        />
                      </span>
                    </div>
                  </ForwardRef>
                </CardTitle>
                <div
                  className="d-flex flex-column"
                >
                  <span
                    className="appointmentCardBodyContentPractitioner ellipsis"
                    data-testid="appointment-item__practitioner-card"
                  />
                  <span
                    className="appointmentCardBodyContentTime ellipsis"
                  >
                    0:00 - 0:00 pm
                  </span>
                </div>
              </div>
            </div>
          </CardBody>
        </div>
      </Card>
    </div>
  </Col>
</AppointmentItem>
`;
